### PR TITLE
feat: Design PayTM gateway modals

### LINKED ISSUE #3446: Prepare Modal Designs for Paytm checkout
**Is your feature request related to a problem? Please describe.**

Modal designs are required for paytm checkout and we have to design them manually.

On pending order page, Link these 2 modals with the "Pay with Paytm" option.

- Paytm Payment Options Modal
  * Should contain one wallet option (for now)
  * Space for entering registered mobile number
- OTP checkout
  * Space for entering received OTP

### PR DESCRIPTION
**Pull request by uds5501**
Fixes #3446

#### Short description of what this resolves:
Add PayTM modal designs. 
(actions will be linked once server PR has been completed)

#### Changes proposed in this pull request:
- PayTM payment options modal created
Screenshot from 2019-08-20 21-01-30

- OTP entering modal prepared simultaneously
Screenshot from 2019-08-20 21-04-13

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mrsaicharan1** [comment]
For each modal, just keep adding screenshots of them to the description 🖼

**Comment by uds5501** [comment]
@mrsaicharan1 Yes, I'll be doing that

File: app/templates/components/modals/paytm-otp.hbs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  {{t 'Enter OTP sent to mobile number'}}
+  <div class="ui {{if isLoading 'loading'}} form" autocomplete="off">
+    {{input type='number' name='otp' value=otp required=true}}
+  </div>
+</div>
+
+<div class="actions">
+  <button class="ui green button" disabled={{isOTPvalid}}>
+    {{t 'Verify'}}
+  </button>
```
Comment (shreyanshdwivedi) [inline app/templates/components/modals/paytm-otp.hbs]: @uds5501 I think that there should be a cancel button for this modal too

Comment (uds5501) [reply app/templates/components/modals/paytm-otp.hbs]: @shreyanshdwivedi I have updated the modal according to the suggestion, please review now

File: app/templates/components/modals/paytm-otp.hbs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<div class="header">
+  {{t 'Amount to be paid:'}} {{currency-symbol currency}} {{amount}}
+</div>
+
+<div class="content">
+  {{t 'Enter OTP sent to mobile number'}}
+  <div class="ui {{if isLoading 'loading'}} form" autocomplete="off">
+    {{input type='number' name='otp' value=otp required=true}}
```
Comment (kushthedude) [inline app/templates/components/modals/paytm-otp.hbs]: Add a validation rule for it

File: app/templates/components/modals/paytm-payment-options.hbs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  </div>
+  <div class="ui {{if isLoading 'loading'}} form" autocomplete="off" >
+    <div class="field">
+      {{ui-radio name='payment_mode' value='paytm' onChange=(action (mut isWalletSelected))}}
+      {{t 'Paytm Wallet'}}<img src="/images/payment-logos/paytm.png" alt="paytm"> 
+    </div>
+    {{#if isWalletSelected}}
+      <div class="field">
+        <div class="label">
+          {{t 'Please enter your Paytm registered Mobile Number to continue'}}
```
Comment (kushthedude) [inline app/templates/components/modals/paytm-payment-options.hbs]: Validation rule

File: app/templates/orders/pending.hbs [hunk 4]
```diff
@@ -61,6 +61,9 @@
             </form>
           </div>
         {{/if}}
+        {{#if isPaytm}}
+          <button class="ui button primary" {{action "openPaytmModal"}}>{{t 'Pay with Paytm'}}</button>
```
Comment (kushthedude) [inline app/templates/orders/pending.hbs]: `Pay with PayTM`

**Comment by uds5501** [comment]
@mrsaicharan1 @shreyanshdwivedi @kushthedude I have added the required validations in the modals. Please review now

File: app/components/modals/paytm-payment-options.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  getValidationRules() {
+    return {
+      inline : true,
+      delay  : false,
+      on     : 'blur',
+      fields : {
+        mobileNumber: {
+          identifier : 'mobile_number',
+          rules      : [
```
Comment (kushthedude) [inline app/components/modals/paytm-payment-options.js]: @uds5501 Please add the `regex` too

Comment (uds5501) [reply app/components/modals/paytm-payment-options.js]: @kushthedude added them

**Pull request merged**